packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -0,0 +1,855 @@
+    );
+  }
+}
+
+enum _WindowControllerAspect { contentSize, title, activated, maximized, minimized, fullscreen }
+
+/// Provides descendants with access to the [BaseWindowController] associated with
+/// the window that is being rendered.
+///
+/// {@macro flutter.widgets.windowing.experimental}
+///
+/// See also:
+///
+///  * [RegularWindow], the widget to create a regular window.
+@internal
+class WindowControllerScope extends InheritedModel<_WindowControllerAspect> {
+  /// Creates a new [WindowControllerScope].
+  ///
+  /// This widget is used by the window widgets to provide
+  /// widgets in a window's subtree with access to information about
+  /// the window.
+  ///
+  /// The [controller] is the controller associated with this window,
+  /// and the [child] is the widget tree that will have access
+  /// to this context.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  WindowControllerScope({super.key, required this.controller, required super.child}) {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_kWindowingDisabledErrorMessage);
+    }
+  }
+
+  /// The controller associated with this window.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  final BaseWindowController controller;
+
+  /// Returns the [WindowControllerScope].
+  ///
+  /// {@template flutter.widgets.windowing.windowControllerScope.of}
+  /// If [isWindowingEnabled] is `false`, this method will throw an
+  /// [UnsupportedError].
+  ///
+  /// If there is no [WindowControllerScope] in scope, this method
+  /// will throw a [TypeError] exception in release builds, and thrown
+  /// a descriptive [FlutterError] in debug builds.
+  /// {@endtemplate}
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  ///
+  /// See also:
+  /// * [RegularWindowController], the controller for regular top-level windows.
+  /// * [RegularWindow], the widget for a regular window.
+  /// * [maybeOf], which doesn't throw or assert if it doesn't find a
+  ///   [WindowControllerScope] ancestor. It returns null instead.
+  @internal
+  static BaseWindowController of(BuildContext context) {
+    return _of(context);
+  }
+
+  /// Returns the [WindowControllerScope] if one exists, otherwise null.
+  ///
+  /// If [isWindowingEnabled] is `false`, this method will throw an
+  /// [UnsupportedError].
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  ///
+  /// See also:
+  /// * [RegularWindowController], the controller for regular top-level windows.
+  /// * [RegularWindow], the widget for a regular window.
+  /// * [of], which will throw if it doesn't find a [WindowControllerScope] ancestor,
+  ///   instead of returning null.
+  @internal
+  static BaseWindowController? maybeOf(BuildContext context) {
+    return _maybeOf(context);
+  }
+
+  /// Returns [BaseWindowController.contentSize] of the nearest [WindowControllerScope].
```

Comment:
Could you add a "the" to this comment and the ones below as well?

```suggestion
  /// Returns the [BaseWindowController.contentSize] of the nearest [WindowControllerScope].
```